src/librustc_mir/interpret/const_eval.rs

```diff
@@ -284,6 +284,28 @@ impl<'mir, 'tcx> super::Machine<'mir, 'tcx> for CompileTimeEvaluator {
                 ecx.write_primval(dest, PrimVal::from_u128(type_id), dest_layout.ty)?;
             }
 
+            "uninit" => {
+                let size = dest_layout.size.bytes();
+                let uninit = |this: &mut EvalContext<'a, 'mir, 'tcx, Self>, val: Value| match val {
+                    Value::ByRef(ptr, ..) => {
+                        this.memory.mark_definedness(ptr, size, false)?;
+                        Ok(val)
+                    }
+                    _ => Ok(Value::ByVal(PrimVal::Undef)),
+                };
+                match dest {
+                    Place::Local { frame, local } => ecx.modify_local(frame, local, uninit)?,
+                    Place::Ptr {
+                        ptr,
```

Comment:
I copy pasted this code from `src/tools/miri`. The miri version had `ptr: PtrAndAlign { ptr, aligned: true }` in the pattern, but that got lost in translation. How do I check that `ptr` is aligned in this context? `Pointer` API didn't have any method to check for alignment AFAICT.

Reply:
Did you use the old code? You could try looking again, now that miri has been updated.